File: server/router/kubernetes/k8s_cluster.go
package kubernetes

import (
	"github.com/gin-gonic/gin"
	v1 "kubespace/server/api/v1"
	"kubespace/server/middleware"
)

type ClusterRouter struct {}

func (c ClusterRouter) InitClusterRouter(r *gin.RouterGroup) {
	K8sClusterRouter := r.Group("k8s").Use(middleware.OperationRecord())
	k8s := v1.ApiGroupApp.KubernetesApiGroup.ClusterApi
	{
		K8sClusterRouter.POST("cluster", k8s.CreateK8SCluster)
		K8sClusterRouter.GET("cluster/secret", k8s.ClusterSecret)
		K8sClusterRouter.POST("cluster/delete", k8s.DelK8SCluster)
		K8sClusterRouter.GET("cluster", k8s.ListK8SCluster)
		K8sClusterRouter.GET("cluster/detail", k8s.GetK8SClusterDetail)
		K8sClusterRouter.GET("events", k8s.Events)
	}
}

File: web/src/api/k8sCluster.js
import service from '@/utils/request'

export const k8sCluster = (data) => {
     return service({
         url: "/k8s/cluster",
         method: 'post',
         data
     })
 }


 export const fetchK8SCluster = (data) => {
     return service({
         url: "/k8s/cluster",
         method: 'get',
         data
     })
 }


 export const clusterSecret = (data) => {
     return service({
         url: "/k8s/cluster/secret",
         method: 'get',
         data
     })
 }

 export const delK8SCluster = (data) => {
    return service({
        url: "/k8s/cluster/delete",
        method: 'post',
        data
    })
}

export const getK8SClusterDetail = (data) => {
    return service({
        url: "/k8s/cluster/detail",
        method: 'get',
        data
    })
}

export const getEvents = (data) => {
    return service({
        url: "/k8s/events",
        method: 'get',
        data
    })
}
